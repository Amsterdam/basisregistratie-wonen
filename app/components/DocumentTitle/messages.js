import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  default_title: {
    id: `${scope}.registration_plural`,
  },
});
