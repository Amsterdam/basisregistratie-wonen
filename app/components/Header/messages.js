/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  logged_in_as: {
    id: `${scope}.logged_in_as`,
  },
  log_in: {
    id: `${scope}.log_in`,
  },
  log_in_adw: {
    id: `${scope}.log_in_adw`,
  },
  log_out: {
    id: `${scope}.log_out`,
  },
});
