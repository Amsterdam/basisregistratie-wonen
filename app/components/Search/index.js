import React, { forwardRef } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Suggest from 'components/Suggest';

const Form = styled.form`
  display: flex;
  justify-content: center;
  width: 100%;

  legend {
    border: 0;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
  }
`;

const Input = styled.input`
  border: #767676 solid 1px;
  height: 44px;
  padding: 9px 9px 11px;
  width: 100%;

  &,
  &::placeholder {
    font-size: 1em;
  }

  &:focus {
    outline: #fec813 solid 4px;
    outline-offset: 0;
    border: black solid 1px;
  }
`;

const StyledSuggest = styled(Suggest)`
  background: white;
  position: absolute;
  border: #767676 solid 1px;
  min-width: 70px;
  max-width: 620px;
  width: 100%;
  top: calc(100% - 4px);
  padding-top: 8px !important;
`;

const Wrapper = styled.div`
  min-width: 70px;
  max-width: 620px;
  width: 100%;
`;

const Search = forwardRef(({ onChange, onSelect, results, placeholder }, ref) => (
  <Form action="">
    <Wrapper>
      <Input
        autoCapitalize="off"
        autoComplete="off"
        autoCorrect="off"
        placeholder={placeholder}
        spellCheck={false}
        type="text"
        defaultValue=""
        onKeyUp={onChange}
        ref={ref}
      />
      {results.length > 0 && <StyledSuggest items={results} onSelect={onSelect} />}
    </Wrapper>
  </Form>
));

Search.defaultProps = {
  placeholder: '',
  results: [],
};

Search.propTypes = {
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSelect: PropTypes.func.isRequired,
  results: PropTypes.arrayOf(PropTypes.shape({})),
};

export default Search;
