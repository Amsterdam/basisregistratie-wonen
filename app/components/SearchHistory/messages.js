/*
 * SearchHistory Messages
 *
 * This contains all the text for the SearchHistory component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  search_history_header: {
    id: `${scope}.search_history_header`,
    defaultMessage: 'Recently search for',
  },
});
