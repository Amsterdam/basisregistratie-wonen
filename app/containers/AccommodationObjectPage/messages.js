/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  yes: {
    id: 'boilerplate.containers.AccommodationObjectPage.yes',
    defaultMessage: 'Ja',
  },
  no: {
    id: 'boilerplate.containers.AccommodationObjectPage.no',
    defaultMessage: 'Nee',
  },
  does_not_apply: {
    id: 'boilerplate.containers.AccommodationObjectPage.does_not_apply',
    defaultMessage: 'n.v.t.',
  },
});
