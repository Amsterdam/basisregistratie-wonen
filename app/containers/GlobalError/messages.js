/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  unable_to_fetch: {
    id: `${scope}.unable_to_fetch`,
  },
  unauthorized: {
    id: `${scope}.unauthorized`,
  },
  session_expired: {
    id: `${scope}.session_expired`,
  },
});
