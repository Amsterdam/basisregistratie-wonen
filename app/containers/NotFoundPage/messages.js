/*
 * NotFoundPage Messages
 *
 * This contains all the text for the NotFoundPage component.
 */
import { defineMessages } from 'react-intl';
import { scope } from '../../i18n';

export default defineMessages({
  document_title: {
    id: `${scope}.registration_plural`,
  },
  header: {
    id: `${scope}.page_not_found`,
  },
});
