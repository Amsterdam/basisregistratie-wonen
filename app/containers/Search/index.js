import React, { useState, useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import { compose, bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { injectIntl, intlShape } from 'react-intl';
import { debounce } from 'lodash';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';
import { pushSearchHistoryLocalStorage } from 'utils/searchHistory';
import Search from 'components/Search';

import { makeSelectResults, makeSelectSuggestionResults, makeIsLoading } from './selectors';
import reducer from './reducer';
import saga from './saga';
import { inputChanged, searchSelect } from './actions';
import { pushSearchHistory } from '../SearchHistory/actions';

import messages from './messages';

export const SearchContainerComponent = props => {
  const inputRef = useRef();
  const suggestRef = useRef();
  const [showSuggest, setShowSuggest] = useState(true);

  useEffect(() => {
    inputRef.current.addEventListener('focusout', onFocusOut, true);
    suggestRef.current.addEventListener('focusout', onFocusOut, true);
    document.addEventListener('click', onClick, true);

    return () => {
      inputRef.current.removeEventListener('focusout', onFocusOut, true);
      suggestRef.current.removeEventListener('focusout', onFocusOut, true);
      document.removeEventListener('click', onClick, true);
    };
  }, []);

  // Bug in Safari, Safari iOS
  // ==========================
  // In Safari the focusout event is triggered before
  // the click is handled. And because the links are hidden,
  // they are not followed.
  // Maybe a better overall solution would be
  // to use React's `onFocus` and `onBlur` vs. refs + focusout.
  const onFocusOut = event => {
    const { relatedTarget } = event;

    const { current: input } = inputRef;
    const { current: suggest } = suggestRef;

    const blurFromInput = !input || (input !== null && !input.contains(relatedTarget));
    const blurFromSuggest = !suggest || (suggest !== null && !suggest.contains(relatedTarget));

    if (blurFromInput && blurFromSuggest) {
      setShowSuggest(false);
    }
  };

  const onFocus = () => {
    setShowSuggest(true);
  };

  const debouncedOnChange = debounce(props.onChange, 200);

  const onChange = event => {
    event.persist();

    const { target } = event;

    const trimmedValue = target.value.trim();
    if (trimmedValue.length < 3) return;

    debouncedOnChange(trimmedValue);
  };

  const onSelect = event => {
    event.preventDefault();
    event.persist();

    const {
      currentTarget: { dataset, text },
    } = event;

    props.pushSearchHistory({ ...dataset, text });
    pushSearchHistoryLocalStorage({ ...dataset, text });

    props.onSearchSelect({ ...dataset });
    inputRef.current.value = text;
  };

  const onSuggestionSelect = event => {
    event.preventDefault();
    event.persist();

    const {
      currentTarget: { text },
    } = event;

    inputRef.current.value = `${text} `;
    props.onChange(`${text} `);
    inputRef.current.focus();
  };

  const onSubmit = event => {
    event.preventDefault();
  };

  const eventOutside = (event, eventType) => {
    const { type, target } = event;
    const { current: input } = inputRef;
    const { current: suggest } = suggestRef;

    const eventOutsideInput = !input || (input !== null && target !== input && !input.contains(target));
    const eventOutsideSuggest = !suggest || (suggest !== null && target !== suggest && !suggest.contains(target));
    return type === eventType && eventOutsideInput && eventOutsideSuggest;
  };

  const onClick = event => {
    const isClickOutside = eventOutside(event, 'click');

    if (isClickOutside && showSuggest !== isClickOutside) {
      setShowSuggest(!isClickOutside);
    }
  };

  const { intl, results, suggestionResults, show, isLoading } = props;
  const visibleResults = showSuggest ? results : {};

  return (
    <Search
      onChange={onChange}
      onFocus={onFocus}
      onSelect={onSelect}
      onSuggestionSelect={onSuggestionSelect}
      onSubmit={onSubmit}
      ref={inputRef}
      suggestRef={suggestRef}
      suggestionResults={suggestionResults}
      results={visibleResults}
      formLegendLabel={intl.formatMessage(messages.search_form_legend)}
      searchTermLabel={intl.formatMessage(messages.search_term)}
      searchHintLabel={intl.formatMessage(messages.search_hint)}
      searchLabel={intl.formatMessage(messages.search)}
      as="form"
      startFoldedOut={show}
      isLoading={isLoading}
    />
  );
};

SearchContainerComponent.defaultProps = {
  results: undefined,
  suggestionResults: undefined,
  show: false,
};

SearchContainerComponent.propTypes = {
  intl: intlShape.isRequired,
  onChange: PropTypes.func.isRequired,
  onSearchSelect: PropTypes.func.isRequired,
  pushSearchHistory: PropTypes.func.isRequired,
  results: PropTypes.shape({}),
  suggestionResults: PropTypes.shape({}),
  show: PropTypes.bool,
  isLoading: PropTypes.bool,
};

const mapStateToProps = createStructuredSelector({
  results: makeSelectResults,
  suggestionResults: makeSelectSuggestionResults,
  isLoading: makeIsLoading,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      onChange: inputChanged,
      onSearchSelect: searchSelect,
      pushSearchHistory,
    },
    dispatch,
  );

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);
const withReducer = injectReducer({ key: 'search', reducer });
const withSaga = injectSaga({ key: 'search', saga });
const Intl = injectIntl(SearchContainerComponent);

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(Intl);
