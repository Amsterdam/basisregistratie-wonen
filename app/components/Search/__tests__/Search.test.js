import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import { Provider } from 'react-redux';
import history from 'utils/history';
import 'jest-styled-components';
import { ThemeProvider } from '@datapunt/asc-ui';
import { IntlProvider } from 'react-intl';
import Search from '..';
import configureStore from '../../../configureStore';
import messages from '../../../translations/nl.json';

const store = configureStore({}, history);
const intlProvider = new IntlProvider({ locale: 'nl', messages });
const { intl } = intlProvider.getChildContext();

describe('Search', () => {
  const results = {
    Adressen: [
      {
        name: 'Foo bar',
        vboId: '0363010000740956',
      },
      {
        name: 'Bar baz',
        vboId: '0363010000740957',
      },
    ],
  };

  const testProps = {
    onChange: () => {},
    onSelect: () => {},
    onSuggestionSelect: () => {},
    formLegendLabel: 'Legend text',
    searchLabel: 'Toggle text',
    searchTermLabel: 'Term text',
    searchHintLabel: 'Hint text',
  };

  const renderSearch = props => (
    <ThemeProvider>
      <Provider store={store}>
        <IntlProvider locale="nl" messages={messages} intl={intl}>
          <Search {...props} />
        </IntlProvider>
      </Provider>
    </ThemeProvider>
  );

  afterEach(cleanup);

  it('matches the snapshot', () => {
    const { container, getByText } = render(renderSearch(testProps));

    expect(container.firstChild).toMatchSnapshot();
    expect(getByText('Legend text')).not.toBeUndefined();
    expect(getByText('Toggle text')).not.toBeUndefined();
    expect(getByText('Term text')).not.toBeUndefined();
    expect(getByText('Hint text')).not.toBeUndefined();

    // it should hide the search from print view and not display the legend
    expect(container.firstChild.classList).toContain('no-print');
    expect(container.getElementsByTagName('legend')[0].classList).toContain('visuallyhidden');
  });

  it('should render results', () => {
    const { container, getByText } = render(renderSearch({ ...testProps, results }));

    expect(container.firstChild).toMatchSnapshot();

    expect(getByText('Adressen')).not.toBeUndefined();
    expect(getByText('Foo bar')).not.toBeUndefined();
    expect(getByText('Bar baz')).not.toBeUndefined();
  });

  it('should show the search form', () => {
    const foldoutProps = { ...testProps, startFoldedOut: true };
    const { getByTestId } = render(renderSearch(foldoutProps));
    const foldOut = getByTestId('search-foldout');

    expect(foldOut).toHaveStyleRule('display', 'block');
  });

  it('should toggle the display of the form', () => {
    const { getByTestId } = render(renderSearch(testProps));

    const foldOut = getByTestId('search-foldout');

    expect(foldOut).toHaveStyleRule('display', 'none');

    // disabling error(), because jsdom outputs the error 'Not implemented: HTMLFormElement.prototype.submit' when
    // click events in the form bubble up to the actual HTMLFormElement
    global.console.error = jest.fn();

    fireEvent(
      getByTestId('search-toggle'),
      new MouseEvent('click', {
        bubbles: true,
      }),
    );

    expect(foldOut).toHaveStyleRule('display', 'block');

    fireEvent(
      getByTestId('search-toggle'),
      new MouseEvent('click', {
        bubbles: true,
      }),
    );

    expect(foldOut).toHaveStyleRule('display', 'none');

    global.console.error.mockRestore();
  });

  it('should call the onFocus handler', () => {
    const onFocus = jest.fn();
    render(renderSearch({ ...testProps, onFocus }));

    document.getElementById('searchInput').focus();

    expect(onFocus).toHaveBeenCalled();
  });

  it('should call the onChange handler', () => {
    const onChange = jest.fn();
    render(renderSearch({ ...testProps, onChange }));

    fireEvent.change(document.getElementById('searchInput'), { target: { value: 'a' } });

    expect(onChange).toHaveBeenCalled();
  });

  it('should call the onSelect handler', () => {
    const onSelect = jest.fn();
    const { getByText } = render(renderSearch({ ...testProps, onSelect, results }));

    const anchor = getByText('Foo bar').closest('a');

    fireEvent(
      anchor,
      new MouseEvent('click', {
        bubbles: true,
      }),
    );

    expect(onSelect).toHaveBeenCalled();
  });

  it('should render loadingIndicator', () => {
    const { queryByTestId } = render(renderSearch({ ...testProps, isLoading: true }));
    expect(queryByTestId('progress-wrapper')).toBeTruthy();
  });

  it('should not render loadingIndicator', () => {
    const { queryByTestId } = render(renderSearch({ ...testProps, isLoading: false }));
    expect(queryByTestId('progress-wrapper')).toBeNull();
  });
});
