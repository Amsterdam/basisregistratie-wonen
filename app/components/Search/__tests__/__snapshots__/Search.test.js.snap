// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches the snapshot 1`] = `
.c2 {
  padding: 10px;
  z-index: 1;
  width: 50px;
  height: 50px;
  border: 0;
  background: none;
}

.c2:hover {
  background-color: #ec0000;
  cursor: pointer;
}

.c2:hover svg {
  fill: white;
}

.c2:focus {
  outline: 4px solid #fec813;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  background-color: #f5f5f5;
}

.c6 {
  font-size: 18px;
  line-height: 25px;
  font-family: Avenir Next LT W01 Demi;
  font-weight: 400;
  margin-bottom: 10px;
}

.c8 {
  border: #767676 solid 1px;
  height: 44px;
  padding: 9px 9px 11px;
  width: 100%;
  margin: 10px 0;
}

.c8,
.c8::-webkit-input-placeholder {
  font-size: 1em;
}

.c8,
.c8::-moz-placeholder {
  font-size: 1em;
}

.c8,
.c8:-ms-input-placeholder {
  font-size: 1em;
}

.c8,
.c8::placeholder {
  font-size: 1em;
}

.c8:focus {
  outline: #fec813 solid 4px;
  outline-offset: 0;
  border: black solid 1px;
}

.c7 {
  font-size: 16px;
  line-height: 22px;
  color: #767676;
}

.c1 {
  max-width: 1024px;
  width: 100%;
  z-index: 1;
  position: relative;
  margin-bottom: 0px;
}

.c4 {
  min-width: 70px;
  max-width: 620px;
  margin: 0 auto;
  position: relative;
}

.c9 {
  position: absolute;
  min-width: 70px;
  max-width: 620px;
  width: 100%;
  top: calc(100% - 11px);
  z-index: 1;
}

.c3 {
  position: absolute;
  right: 0;
  top: -50px;
}

.c5 {
  display: none;
  max-width: 620px;
  margin: 0 auto;
}

@media (max-width:1200px) {
  .c0 {
    padding-top: 54px;
  }
}

@media (max-width:650px) {
  .c4 {
    max-width: calc(100vw - 30px);
  }
}

@media (max-width:650px) {
  .c9 {
    max-width: calc(100vw - 30px);
  }
}

<form
  action=""
  class="c0 no-print"
  data-testid="search-form"
>
  <legend
    class="visuallyhidden"
  >
    Legend text
  </legend>
  <div
    class="c1"
  >
    <button
      class="c2 c3"
      data-testid="search-toggle"
      size="20"
      type="button"
    >
      <svg
        data-testid="searchToggle-search-icon"
        focusable="false"
        viewBox="0 0 32 32"
        width="20"
      >
        <path
          d="M32 29.119l-9.825-9.825a12.217 12.217 0 1 0-2.88 2.881L29.118 32 32 29.119zm-19.771-8.746a8.145 8.145 0 1 1 0-16.289 8.145 8.145 0 0 1 0 16.29z"
          fill-rule="evenodd"
        />
      </svg>
      <span
        class="visuallyhidden"
        data-testid="searchToggle-label"
      >
        Toggle text
      </span>
    </button>
    <div
      class="c4"
    >
      <div
        class="c5"
        data-testid="search-foldout"
      >
        <label
          class="c6"
          for="searchInput"
        >
          Term text
        </label>
        <div
          class="c7"
        >
          Hint text
        </div>
        <input
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="c8"
          id="searchInput"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
      <div
        class="c9"
      />
    </div>
  </div>
</form>
`;

exports[`Search should render results 1`] = `
.c12 {
  margin: 0;
  color: #000000;
  font-weight: inherit;
  font-size: 1rem;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.25;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 700;
  color: #000000;
}

.c13:focus {
  background-color: #FEC813;
}

.c13 .c17 {
  margin: 3px;
}

.c13 .c17 svg rect,
.c13 .c17 svg polygon,
.c13 .c17 svg path {
  fill: #000000;
}

.c13:hover {
  color: #ec0000;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c13:hover .c17 svg rect,
.c13:hover .c17 svg polygon,
.c13:hover .c17 svg path {
  fill: #ec0000;
}

.c2 {
  padding: 10px;
  z-index: 1;
  width: 50px;
  height: 50px;
  border: 0;
  background: none;
}

.c2:hover {
  background-color: #ec0000;
  cursor: pointer;
}

.c2:hover svg {
  fill: white;
}

.c2:focus {
  outline: 4px solid #fec813;
}

.c14 {
  padding: 0 !important;
  font-weight: 400;
}

.c14:focus {
  outline: 4px solid #ffc90e !important;
}

.c14:hover,
.c14:focus {
  color: #ec0000;
  fill: #ec0000;
}

.c16 {
  vertical-align: baseline;
  margin-right: 5px;
}

.c11 {
  padding-left: 4px;
}

.c15 {
  margin-left: 4px !important;
  line-height: 26px;
}

.c15 span {
  vertical-align: top;
}

.c10 {
  background: white;
  border-top: none;
  border: #767676 solid 1px;
  width: 100%;
}

.c10 li:first-of-type {
  margin-top: 8px !important;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  background-color: #f5f5f5;
}

.c6 {
  font-size: 18px;
  line-height: 25px;
  font-family: Avenir Next LT W01 Demi;
  font-weight: 400;
  margin-bottom: 10px;
}

.c8 {
  border: #767676 solid 1px;
  height: 44px;
  padding: 9px 9px 11px;
  width: 100%;
  margin: 10px 0;
}

.c8,
.c8::-webkit-input-placeholder {
  font-size: 1em;
}

.c8,
.c8::-moz-placeholder {
  font-size: 1em;
}

.c8,
.c8:-ms-input-placeholder {
  font-size: 1em;
}

.c8,
.c8::placeholder {
  font-size: 1em;
}

.c8:focus {
  outline: #fec813 solid 4px;
  outline-offset: 0;
  border: black solid 1px;
}

.c7 {
  font-size: 16px;
  line-height: 22px;
  color: #767676;
}

.c1 {
  max-width: 1024px;
  width: 100%;
  z-index: 1;
  position: relative;
  margin-bottom: 0px;
}

.c4 {
  min-width: 70px;
  max-width: 620px;
  margin: 0 auto;
  position: relative;
}

.c9 {
  position: absolute;
  min-width: 70px;
  max-width: 620px;
  width: 100%;
  top: calc(100% - 11px);
  z-index: 1;
}

.c3 {
  position: absolute;
  right: 0;
  top: -50px;
}

.c5 {
  display: none;
  max-width: 620px;
  margin: 0 auto;
}

@media screen and (-ms-high-contrast:active),(-ms-high-contrast:none) {
  .c10 a {
    height: 26px;
  }
}

@media (max-width:1200px) {
  .c0 {
    padding-top: 54px;
  }
}

@media (max-width:650px) {
  .c4 {
    max-width: calc(100vw - 30px);
  }
}

@media (max-width:650px) {
  .c9 {
    max-width: calc(100vw - 30px);
  }
}

<form
  action=""
  class="c0 no-print"
  data-testid="search-form"
>
  <legend
    class="visuallyhidden"
  >
    Legend text
  </legend>
  <div
    class="c1"
  >
    <button
      class="c2 c3"
      data-testid="search-toggle"
      size="20"
      type="button"
    >
      <svg
        data-testid="searchToggle-search-icon"
        focusable="false"
        viewBox="0 0 32 32"
        width="20"
      >
        <path
          d="M32 29.119l-9.825-9.825a12.217 12.217 0 1 0-2.88 2.881L29.118 32 32 29.119zm-19.771-8.746a8.145 8.145 0 1 1 0-16.289 8.145 8.145 0 0 1 0 16.29z"
          fill-rule="evenodd"
        />
      </svg>
      <span
        class="visuallyhidden"
        data-testid="searchToggle-label"
      >
        Toggle text
      </span>
    </button>
    <div
      class="c4"
    >
      <div
        class="c5"
        data-testid="search-foldout"
      >
        <label
          class="c6"
          for="searchInput"
        >
          Term text
        </label>
        <div
          class="c7"
        >
          Hint text
        </div>
        <input
          autocapitalize="off"
          autocomplete="off"
          autocorrect="off"
          class="c8"
          id="searchInput"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
      <div
        class="c9"
      >
        <ul
          class="c10  links"
        >
          <li>
            <strong
              class="c11"
            >
              Adressen
            </strong>
            <ul
              class="links"
            >
              <li>
                <a
                  class="c12 c13 c14 c15"
                  data-vbo-id="0363010000740956"
                  href="/"
                >
                  <svg
                    class="c16"
                    focusable="false"
                    viewBox="0 0 32 32"
                    width="14"
                  >
                    <path
                      d="M9.757 32l-2.9-2.91L19.937 16 6.857 2.91 9.757 0l16 16z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <span
                    class="linklabel"
                  >
                    Foo bar
                  </span>
                </a>
              </li>
              <li>
                <a
                  class="c12 c13 c14 c15"
                  data-vbo-id="0363010000740957"
                  href="/"
                >
                  <svg
                    class="c16"
                    focusable="false"
                    viewBox="0 0 32 32"
                    width="14"
                  >
                    <path
                      d="M9.757 32l-2.9-2.91L19.937 16 6.857 2.91 9.757 0l16 16z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <span
                    class="linklabel"
                  >
                    Bar baz
                  </span>
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </div>
</form>
`;
